Mathlib/Combinatorics/SetFamily/Shadow.lean

```diff
@@ -104,13 +107,52 @@ theorem mem_shadow_iff_insert_mem : s ∈ ∂ 𝒜 ↔ ∃ (a : _) (_ : a ∉ s)
     exact ⟨insert a s, hs, a, mem_insert_self _ _, erase_insert ha⟩
 #align finset.mem_shadow_iff_insert_mem Finset.mem_shadow_iff_insert_mem
 
+/-- `t ∈ ∂𝒜` iff `t` is exactly one element less than something from `𝒜` -/
+lemma mem_shadow_iff_exists_sdiff : t ∈ ∂ 𝒜 ↔ ∃ s ∈ 𝒜, t ⊆ s ∧ (s \ t).card = 1 := by
+  simp_rw [mem_shadow_iff_insert_mem, card_eq_one]
+  constructor
+  · rintro ⟨i, hi, ht⟩
+    exact ⟨insert i t, ht, subset_insert _ _, i, insert_sdiff_cancel hi⟩
+  · rintro ⟨s, hs, hts, a, ha⟩
+    refine' ⟨a, (mem_sdiff.1 $ (ext_iff.1 ha _).2 $ mem_singleton_self _).2, _⟩
+    rwa [insert_eq, ←ha, sdiff_union_of_subset hts]
```

Comment:
Is there an `sdiff_card_eq_one` lemma hiding here, with conclusion ∃ (a : _) (_ : a ∉ t), insert a t = s`?

Reply:
There's a bunch of equivalences here, eg `t = s.erase a`, `s` covers `t`... Do I need a TFAE statement?

Reply:
I think the covby statement is the right one for. `card_sdiff_eq_one`, good point.

Reply:
This worked out nicely.